openpype/plugins/publish/integrate_new.py

```diff
@@ -113,816 +159,538 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
+            # from the source indexes, etc.
+            first_index_padded = get_frame_padded(frame=destination_indexes[0],
+                                                  padding=destination_padding)
+            if is_udim:
+                # UDIM representations handle ranges in a different manner
+                template_data["udim"] = first_index_padded
+            else:
+                template_data["frame"] = first_index_padded
+
+            # Construct destination collection from template
+            anatomy_filled = anatomy.format(template_data)
+            template_filled = anatomy_filled[template_name]["path"]
+            repre_context = template_filled.used_values
+            self.log.debug("Template filled: {}".format(str(template_filled)))
+            dst_collection = assemble([os.path.normpath(template_filled)])
+
+            # Update the destination indexes and padding
+            dst_collection.indexes.clear()
+            dst_collection.indexes.update(set(destination_indexes))
+            dst_collection.padding = destination_padding
+            assert (
+                len(src_collection.indexes) == len(dst_collection.indexes)
+            ), "This is a bug"
+
+            # Multiple file transfers
+            transfers = []
+            for src_file_name, dst in zip(src_collection, dst_collection):
+                src = os.path.join(stagingdir, src_file_name)
+                transfers.append((src, dst))
 
-        try:
-            filled_template = StringTemplate.format_strict_template(
-                template, fill_pairs
+        else:
+            # Single file
+            fname = files
+            assert not os.path.isabs(fname), (
+                "Given file name is a full path"
             )
-        except (KeyError, TemplateUnsolved):
-            keys = []
-            if fill_pairs:
-                keys = fill_pairs.keys()
 
-            msg = "Subset grouping failed. " \
-                  "Only {} are expected in Settings".format(','.join(keys))
-            self.log.warning(msg)
+            # Manage anatomy template data
+            template_data.pop("frame", None)
+            if is_udim:
+                template_data["udim"] = repre["udim"][0]
+
+            # Construct destination filepath from template
+            anatomy_filled = anatomy.format(template_data)
+            template_filled = anatomy_filled[template_name]["path"]
+            repre_context = template_filled.used_values
+            dst = os.path.normpath(template_filled)
+
+            # Single file transfer
+            src = os.path.join(stagingdir, fname)
+            transfers = [(src, dst)]
+
+        # todo: Are we sure the assumption each representation
+        #       ends up in the same folder is valid?
+        if not instance.data.get("publishDir"):
+            instance.data["publishDir"] = (
+                anatomy_filled
+                [template_name]
+                ["folder"]
+            )
```

Comment:
Can anyone confirm the question in this todo? This code is new after a recent pull of latest `develop` with merge conflicts.

Reply:
I believe it should be the case. Don't think there is a way to define where else representation might go.